src/Ucommerce.Transactions.Payments.QuickpayLink/Pipelines/Initialize/SetupQuickpayLinkPaymentMethodDefinitionTask.cs

```diff
@@ -0,0 +1,168 @@
+                definitionRepository ?? throw new ArgumentNullException(nameof(definitionRepository));
+            _definitionFieldRepository = definitionFieldRepository
+                                         ?? throw new ArgumentNullException(nameof(definitionFieldRepository));
+            _definitionTypeRepository = definitionTypeRepository
+                                        ?? throw new ArgumentNullException(nameof(definitionTypeRepository));
+            _dataTypeRepository = dataTypeRepository ?? throw new ArgumentNullException(nameof(dataTypeRepository));
+            _loggingService = loggingService ?? throw new ArgumentNullException(nameof(loggingService));
+        }
+
+        public PipelineExecutionResult Execute(InitializeArgs subject)
+        {
+            CreateQuickpayLinkDefinitions();
+            return PipelineExecutionResult.Success;
+        }
+
+        protected virtual void CreateQuickpayLinkDefinitions()
+        {
+            var sessionProvider = ObjectFactory.Instance.Resolve<ISessionProvider>();
+
+            if (sessionProvider.GetSession() is not SessionImpl session)
+            {
+                throw new Exception("Session not found");
+            }
+
+            var oldListeners = session.Listeners.PreInsertEventListeners;
+            session.Listeners.PreInsertEventListeners = new List<IPreInsertEventListener>().ToArray();
+
+            try
+            {
+                var definition = GetAdyenDefinition();
+                var definitionType = GetPaymentDefinitionType();
+                if (definitionType is null)
+                {
+                    throw new Exception("Payment definition type does not exist");
+                }
+
+                if (definition is null)
+                {
+                    definition = new Definition
+                    {
+                        BuiltIn = true,
+                        CreatedBy = "System",
+                        CreatedOn = DateTime.Now,
+                        ModifiedBy = "System",
+                        ModifiedOn = DateTime.Now,
+                        Description = "Configuration for QuickpayLink",
+                        DefinitionType = definitionType,
+                        Guid = Guid.NewGuid(),
+                        Name = "QuickpayLink"
+                    };
+
+                    _definitionRepository.Save(definition);
+                    definition = GetAdyenDefinition();
+                    if (definition is null)
+                    {
+                        throw new Exception("Unable to create new definition for QuickpayLink");
+                    }
+                }
+
+                CreateNewFields(definition);
+
+                _definitionRepository.Save(definition);
+            }
+            finally
+            {
+                session.Listeners.PreInsertEventListeners = oldListeners;
+            }
+        }
+
+        protected virtual Definition? GetAdyenDefinition()
```

Comment:
This method seems to be named for another payment provider :-)

Maybe a better name would be GetQuickpayLinkDefinition?

Reply:
Renamed to GetQuickpayLinkDefinition.